_posts/2025-04-30-#31981.md

```diff
@@ -7,20 +7,55 @@ authors: [Sjors]
 components: ["mining"]
 host: ismaelsadeeq
 status: upcoming
-commit:
+commit: e6e170cf6c67a56b9c14cece66fdc4fab5f3ec6b
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### Background
+
+[StratumV2 Sv2](https://github.com/stratum-mining/sv2-spec/tree/main) is a standard protocol that improves upon the StratumV1 protocol. 
+Sv2 has several benefits, one of which is allowing individual miners connected to a mining pool to build their own block templates. This further decentralizes Bitcoin mining and improves the censorship resistance of Bitcoin transactions.
+
+The Sv2 standard defines various roles, one of which is the Template Provider (TP), explained in the [template distribution](https://github.com/stratum-mining/sv2-spec/blob/main/07-Template-Distribution-Protocol.md) section of the spec. This role pushes block template updates to miners whenever the chain tip is updated or when mempool fees have increased sufficiently.
+In the Sv2 spec, it is specified that Bitcoin Core should fulfill the TP role, which means it has to implement the Sv2 spec in order to communicate with other roles.
```

Comment:
```suggestion
In the Sv2 spec, it is specified that Bitcoin Core should fulfill the TP role
```

I don't think that's correct?

Reply:
Yes, fixed.

Reply:
Ugh sorry I messed up my comment, I was meant to quote "In the Sv2 spec, it is specified that Bitcoin Core should fulfill the TP role" instead of suggesting it as a fix. My issue was that per my understanding the Sv2 spec does _not_ mandate Bitcoin Core to fulfill the TP role? Anyone should be able to fulfill that role?

Reply:
Yes does not mandate that.
But recommended it, I will soften the tone.

Reply:
Fixed, thanks